core/src/findlib/bfile.cc

```diff
@@ -1185,10 +1186,12 @@ int bclose(BareosWinFilePacket* bfd)
     bfd->cmd_plugin = false;
   } else {
 #if defined(HAVE_POSIX_FADVISE) && defined(POSIX_FADV_DONTNEED)
-    if (bfd->flags_ & O_RDONLY) {
+    /* If not RDWR or WRONLY must be Read Only */
+    if (!(bfd->m_flags & (O_RDWR|O_WRONLY))) {
       fdatasync(bfd->fid); /* sync the file */
```

Comment:
```suggestion
```

Reply:
We discussed this and we agree that calling `fdatasync()` is not a good idea in this case.